src/main/java/net/dv8tion/jda/api/entities/PrivateChannel.java

```diff
@@ -27,8 +31,29 @@
     /**
      * The {@link net.dv8tion.jda.api.entities.User User} that this {@link net.dv8tion.jda.api.entities.PrivateChannel PrivateChannel} communicates with.
      *
-     * @return A non-null {@link net.dv8tion.jda.api.entities.User User}.
+     * <p>This user is only null if this channel is currently uncached, and one the following occur:
+     * <ul>
+     *     <li>A reaction is removed</li>
+     *     <li>A reaction is added</li>
+     *     <li>A message is deleted</li>
+     *     <li>This account sends a message to a user from another shard (not shard 0)</li>
+     * </ul>
+     * The consequence of this is that for any message this bot receives from a guild or from other users will not be null.
+     *
+     *
+     * @return Possibly-null {@link net.dv8tion.jda.api.entities.User User}.
      */
-    @Nonnull
+    @Nullable
     User getUser();
```

Comment:
Add a reference to `retrieveUser()` on this method, it isn't easy to discover otherwise.